packages/webapp-libs/webapp-tenants/src/providers/currentTenant/currentTenant.component.tsx

```diff
@@ -0,0 +1,62 @@
+import { TenantType } from '@sb/webapp-api-client/constants';
+import { reportError } from '@sb/webapp-core/utils/reportError';
+import { prop } from 'ramda';
+import { useMemo } from 'react';
+import { useParams } from 'react-router-dom';
+
+import { useTenants } from '../../hooks/useTenants/useTenants.hook';
+import currentTenantContext from './currentTenant.context';
+import { CURRENT_TENANT_STORAGE_KEY, CurrentTenantProviderProps } from './currentTenant.types';
+
+export type TenantPathParams = {
+  tenantId: string;
+};
+
+/**
+ *
+ * @param children
+ * @constructor
+ *
+ * @category Component
+ */
+export const CurrentTenant = ({ children, storageKey = CURRENT_TENANT_STORAGE_KEY }: CurrentTenantProviderProps) => {
+  const params = useParams<TenantPathParams>();
+  let { tenantId } = params;
+
+  const tenants = useTenants();
+
+  if (!tenantId) {
+    const localId = localStorage.getItem(storageKey);
```

Comment:
I seriously am not a fan of reading and writing to local storage in a render function.

I believe it's a good use case for the `useSyncExternalStore` hook:
- local storage is treated as a single source of truth
- when you want to change value of current tenant you only modify the local storage value (and the set function dispatches an event)
- synchronisation to "react state" is done with `useSyncExternalStore`, which is subscribed to the event mentioned above